controller/array_action/svc_cli_result_reader.py

```diff
@@ -0,0 +1,117 @@
+from controller.common.csi_logger import get_stdout_logger
+import controller.array_action.errors as controller_errors
+
+logger = get_stdout_logger()
+
+ID_PARAM_NAME = "id"
+
+
+class SVCListResultsReader:
+    """
+    Iterable object used to read raw command response from SVC array.
+    Each object in response is translated to SVCListResultsElement.
+    Input is received as string which represents '\n'-separated list of returned lines from output.
+    Line with param 'id' (e.g. 'id 1') is recognized as first line of object ans used as separator between objects
+    (e.g. in input "id 1\nname n3\nid 2\nname n2" first object starts with line 'id 1' and ends with 'id 2')
+    """
+    def __init__(self, hosts_raw_list_as_string):
+        self._hosts_raw_list = hosts_raw_list_as_string.splitlines()
+        self._current_index = 0
+        self._next_object_id = None
+        self._init_first_object_id()
+
+    def _init_first_object_id(self):
+        """
+        Set _init_first_object_id to id o the first object and _current_index to point to next line
```

Comment:
did you mean `id 0`? if so, does it have to be 0?

Reply:
no
there may be host without id 0

Reply:
I still don't understand this line.
what does "Set _init_first_object_id" mean? `_init_first_object_id` is the function name, not a data member

Reply:
@roysahar-ibm yes typo. Fixing

Reply:
Fixed the comment